src/modules/site/site.controller.ts

```diff
@@ -37,6 +43,47 @@ export class SiteController {
   @ApiBadRequestResponse({ description: 'Bad request.' })
   async getSiteStatusByExporterName(@Query() query: GetSiteStatusByExporterNameQueryDto, @Res() res: Response): Promise<void> {
     const getSiteStatusByExporterNameResponse = await this.service.getSiteStatusByExporterName(query.exporterName);
+    this.setSiteStatusResponse(res, getSiteStatusByExporterNameResponse);
+  }
+
+  @Post()
+  @ApiOperation({ summary: 'Start creation of new sharepoint site for exporter' })
+  @ApiOkResponse({ description: 'Site has been created', type: CreateSiteResponse, isArray: false })
+  @ApiAcceptedResponse({ description: 'Site is provisioning', type: CreateSiteResponse, isArray: false })
+  @ApiResponse({
+    status: HttpStatusCode.FailedDependency,
+    description: 'Site has failed to be created',
+    type: CreateSiteResponse,
+    isArray: false,
+  })
+  @ApiUnauthorizedResponse({ description: 'Unauthorized' })
+  @ApiInternalServerErrorResponse({ description: 'An internal server error has occurred.' })
+  @ApiBadRequestResponse({ description: 'Bad request.' })
+  async createSite(@ValidatedArrayBody({ items: CreateSiteRequestItem }) createSiteDto: CreateSiteRequest, @Res() res: Response): Promise<void> {
```

Comment:
Should the return type be `Promise<CreateSiteResponse>` instead?

Reply:
To set http status we need to change object res. Same object is used for response body. It is same approach as in GET site endpoint.

So this function directly is not returning anything.